data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -24,15 +42,96 @@
 public class KafkaSource implements Source<Record<Object>> {
 
   private static final Logger LOG = LoggerFactory.getLogger(KafkaSource.class);
-  
+  private final KafkaSourceConfig sourceConfig;
+  private Set<MultithreadedConsumer> consumers = new HashSet<>();
```

Comment:
I dont see this hash set being used anywhere other than adding consumers to it.

Reply:
resolved